### PR TITLE
GODRIVER-3096 Fixed perfomance CI break by running benchmark pkg not main

### PR DESCRIPTION
**Pull request by prestonvasquez**
GODRIVER-3096

## Summary

Downgrade benchmark to minimum supported version Go 1.18, add benchmark to the workspace.

## Background & Motivation

The changes made in https://github.com/mongodb/mongo-go-driver/commit/b524a7e99d7955920537d87f54a6c53db913b45a no longer work with the performance CI test. Additionally, we should be using the minimum supported version 1.18.

### CODE REVIEW

**Review by timothy-kim-mongo — COMMENTED** [review]
Reran failed test for 8.0 RHEL [test-sharded-auth-ssl-zlib-compression](https://spruce.mongodb.com/task/mongo_go_driver_tests_rhel_42_plus_zlib_zstd_support__version~8.0_os_ssl_40~rhel87_64_go_1_20_test_sharded_auth_ssl_zlib_compression_patch_b79eac454a0c1d3c8e7c55ff723455b6af8cc5d2_669348a6ed8a8e00075ddb64_24_07_14_03_40_23?execution=0) which passed after rerun. LGTM!

**Review by timothy-kim-mongo — DISMISSED** [review]

File: internal/cmd/benchmark/go.mod
Comment (timothy-kim-mongo) [inline internal/cmd/benchmark/go.mod:1]:
```
Compile Check Error found: [2024/07/15 19:08:02.423] ../../../../../../pkg/mod/github.com/klauspost/compress@v1.17.8/zstd/dict.go:492:32: binary.LittleEndian.AppendUint32 undefined (type binary.littleEndian has no field or method AppendUint32)
[2024/07/15 19:08:02.423] note: module requires Go 1.20
```
from [Evergreen Patch Error](https://spruce.mongodb.com/task/mongo_go_driver_build_check_go1.18_build_patch_dbbbae75b9b8ceed1e8c270632573eac83716c8b_66957066a082750007c61c57_24_07_15_18_54_35/logs?execution=0).

It seems from this error that the compress package used at v1.17.8 only works for Go 1.20. I suggest to change the version to `github.com/klauspost/compress v1.13.6` used in GODRIVER v1 which is compatible with Go 1.18 or a possible later version that can work with go version 1.18.

**Review by timothy-kim-mongo — CHANGES_REQUESTED** [review]

File: internal/cmd/benchmark/go.mod
Comment (prestonvasquez) [reply internal/cmd/benchmark/go.mod:1]: Great catch!

**Review by timothy-kim-mongo — APPROVED** [review]
LGTM!

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

**Pull request merged**